CMakeLists.txt

```diff
@@ -602,6 +602,14 @@ else()
   add_subdirectory(Externals/liblzma)
 endif()
 
+find_package(zstd)
+if(ZSTD_FOUND)
+  message(STATUS "Using shared zstd")
+else()
+  message(STATUS "Shared zstd not found, falling back to the static library")
```

Comment:
After rebasing to #8806, you will need `check_vendoring_approved(curl)` here as well.